.github/workflows/ci.yml

```diff
@@ -0,0 +1,176 @@
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - id: set-checks-profiles
+        name: Set CI profiles
+        run: |
+          #!/usr/bin/env bash
+          CI_CHECKS=()
+          HAS_LABEL_CI_NONE=${{ contains(github.event.pull_request.labels.*.name, 'ci:none') }}
+          HAS_LABEL_CI_FAST=${{ contains(github.event.pull_request.labels.*.name, 'ci:fast') }}
+          HAS_LABEL_CI_FULL=${{ contains(github.event.pull_request.labels.*.name, 'ci:full') }}
+
+          # set defaults
+          case "${{ github.event_name == 'pull_request' }}" in
+            true)  CI_CHECKS=( check smoke docs );;
+            false) CI_CHECKS=( check smoke docs base extra );;
+          esac
+
+          [[ ${HAS_LABEL_CI_NONE} == true ]] && CI_CHECKS=()
+          [[ ${HAS_LABEL_CI_FAST} == true ]] && CI_CHECKS=( check smoke docs )
+          [[ ${HAS_LABEL_CI_FULL} == true ]] && CI_CHECKS=( check smoke docs base extra )
+
+          json_result=$(printf ', "%s"' "x" "${CI_CHECKS[@]}")
+          json_result="[${json_result:6}]"
+
+          printf 'checks=%s\n' "${json_result}" >> "${GITHUB_OUTPUT}"
+
+      - id: set-deploy-profiles
+        name: Set CD profiles
+        run: |
+          #!/usr/bin/env bash
+          CD_DEPLOYEMENTS=()
+
+          if [[ ${{ github.event_name == 'push' && (github.ref == 'refs/heads/main') }} == true ]]; then
+            CD_DEPLOYEMENTS+=( docs )
+          fi
+
+          json_result=$(printf ', "%s"' "x" "${CD_DEPLOYEMENTS[@]}")
+          json_result="[${json_result:6}]"
+
+          printf 'deploy=%s\n' "${json_result}" >> "${GITHUB_OUTPUT}"
+
+      - id: print-configuration
+        name: print CI/CD profiles
+        run: |
+          #!/usr/bin/env bash
+          printf 'selected CI profiles: %s\n' '${{ steps.set-checks-profiles.outputs.checks }}'
+          printf 'selected CD profiles: %s\n' '${{ steps.set-deploy-profiles.outputs.deploy }}'
+          printf 'triggering event:\n'
+          cat "${GITHUB_EVENT_PATH}"
+
+  check:
+    needs: [configure]
+    if: ${{ contains(fromJson(needs.configure.outputs.checks), 'check') }}
+    uses: ./.github/workflows/run-checks.yml
+
+  smoke-test:
+    needs: [configure]
+    if: ${{ contains(fromJson(needs.configure.outputs.checks), 'smoke') }}
+    uses: ./.github/workflows/run-smoke-test.yml
+
+  docs:
+    needs: [configure]
+    if: ${{ contains(fromJson(needs.configure.outputs.checks), 'docs') }}
+    uses: ./.github/workflows/build-docs.yml
+
+  run-benchmarks:
+    needs: [configure, smoke-test]
+    if: ${{ contains(fromJson(needs.configure.outputs.checks), 'extra') }}
+    uses: ./.github/workflows/run-benchmarks.yml
+
+  test-tlspuffin:
+    needs: [configure, smoke-test]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: openssl111
+            features: openssl111
+            save-cache: "true" # We only save the cache once, else we get too mache cache entries
```

Comment:
typo

Reply:
Not mine though 😆

@maxammann I'm not sure if it was meant to say that we get _too many_ incompatible cache entries OR that incompatible cache entries were _mashed_ together, resulting in rebuild. Anyway, the caching strategy will change once we can do vendored library builds outside of cargo, through `mk_vendor`. So I guess it will be the right time to remove this comment.

Reply:
Not sure anymore sorry